src/ir_passes/bundle_elim.rs

```diff
@@ -59,6 +56,9 @@ impl BundleElim {
             return vec![pidx];
         }
 
+        // creates an empty info struct for these new ports
+        let info = comp.add(Info::Empty);
```

Comment:
This will start crashing the `ir::Printer` I think because we do the whole:
```
let ir::Info:::Event { ... } = self.get(info) else {
  unreachable!(...)
};
```

I mentioned in the meeting yesterday that we should come up with a better way of extracting `info` objects, either by storing `Option<InfoIdx>` where we store `InfoIdx` or exposing a bunch of methods on `InfoIdx` like `as_event_info()` that return `Option<(...)>` or something.

Also worth looking into the [`struct_variant`](https://docs.rs/struct-variant/latest/struct_variant/) to make it easy to define a diagonstics struct an add it to the `Info` enum.

Reply:
I can work on this probably.